### ISSUE TITLE
Providing {value} as an argument in template for setter results in an error.

### ISSUE THREAD

**Issue by Dia6lo** [bridgedotnet/Bridge#1051]
### Expected
```javascript
var foo = new Foo();
foo.setBar(true);
var baz = foo.currentBar;
```

### Actual
Building throws an error:
```
Error: Input string was not in a correct format.
at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at Bridge.Translator.AbstractEmitterBlock.PopWriter(Boolean preventWrite)
   at Bridge.Translator.AssignmentBlock.VisitAssignmentExpression()
   at Bridge.Translator.AssignmentBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitAssignmentExpression(AssignmentExpression assignmentExpression)
   at ICSharpCode.NRefactory.CSharp.AssignmentExpression.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.ExpressionBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitExpressionStatement(ExpressionStatement expressionStatement)
   at ICSharpCode.NRefactory.CSharp.ExpressionStatement.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.Block.<DoEmitBlock>b__64_0(AstNode child)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Bridge.Translator.Block.DoEmitBlock()
   at Bridge.Translator.Block.EmitBlock()
   at Bridge.Translator.Block.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitBlockStatement(BlockStatement blockStatement)
   at ICSharpCode.NRefactory.CSharp.BlockStatement.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.VisitorMethodBlock.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at Bridge.Translator.VisitorMethodBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitMethodDeclaration(MethodDeclaration methodDeclaration)
   at Bridge.Translator.MethodBlock.EmitMethodsGroup(List`1 group)
   at Bridge.Translator.MethodBlock.EmitMethods(Dictionary`2 methods, Dictionary`2 properties, Dictionary`2 operators)
   at Bridge.Translator.MethodBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.ClassBlock.EmitStaticBlock()
   at Bridge.Translator.ClassBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.EmitBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.Emit()
   at Bridge.Translator.Translator.Translate()
   at Bridge.Build.GenerateScript.Execute()
```

### Steps To Reproduce

```csharp
public class App
{
	[Ready]
	public static void Main()
	{
		var foo = new Foo();
		foo.Bar = true;
		var baz = foo.Bar;
	}
}

[External]
[Name("Foo")]
public class Foo
{
	[FieldProperty]
	[Name("currentBar")]
	public bool Bar
	{
		get { return false; }
		[Template("setBar({value})")]
		set { }
	}
}
```

### Note
Workaround for this issue is changing `[Template("setBar({value})")]` to `[Template("setBar({0})")]`.

**Comment by emumanu** [comment]
@LeonidVeriga , @geoffreymcgill , @vladsch 

why is this closed?

It still doesn't work:

https://deck.net/60d763a8a6f4e4b0e272ad0fc460d2de

**Comment by fabriciomurta** [comment]
This is closed because the scenario described here is not the same as yours. The reported issue here has been fixed by the #1087 pull request referenced. It seems in your case, it involves array indexer. Could you please log it as a new issue?

The scenario reported here matches this deck: https://deck.net/68ca1f281e68b466cf12506d9f108cff

**Comment by emumanu** [comment]
@fabriciomurta , ok. I'm going to open a new issue
